transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java

```diff
@@ -877,7 +848,50 @@ public void testAddReplaceHandlerCalledOnceRegistered() throws Throwable {
         assertTrue(handler2.addedHandler.get());
         assertFalse(handler2.removedHandler.get());
         pipeline.remove(handler2);
-        assertTrue(handler2.removedHandler.get());
+        while (!handler2.removedHandler.get()) {
+            Thread.sleep(10);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)